src/os/bluestore/BlueStore.h

```diff
@@ -765,13 +766,73 @@ class BlueStore : public ObjectStore,
     /// split a blob (and referring extents)
     BlobRef split_blob(BlobRef lb, uint32_t blob_offset, uint32_t pos);
 
-    bool do_write_check_depth(
-      uint64_t onode_size,
-      uint64_t start_offset,
-      uint64_t end_offset,
-      uint8_t  *blob_depth,
-      uint64_t *gc_start_offset,
-      uint64_t *gc_end_offset);
+  };
+
+  ///Compressed Blob Garbage collector
+  /*
+  The primary idea of the collector is to estimate a difference between allocation units(AU)
+  currently present for compressed blobs and new AUs required to store that data uncompressed. 
+  Estimation is performed for protrusive extents within a logical range determined by a concatenation 
+  of old_extents collection and specific(current) write request.
+  The root cause for old_extents use is the need to handle blob ref counts properly. Old extents still hold blob refs
+  and hence we need to traverse the collection to determine if blob to be released.
+  Protrusive extents are extents that fit into the blob set in action(ones that are below the logical range from above)
+  but not removed totally due to the carrent write. e.g. for 
+  extent1 <loffs = 100, boffs = 100, len  = 10> -> blob1<compressed, len_on_disk=4096, logical_len=8192<
+  extent2 <loffs = 200, boffs = 100, len  = 10> -> blob2<raw, len_on_disk=4096, llen=4096>
+  extent3 <loffs = 300, boffs = 100, len  = 10> -> blob1<compressed, len_on_disk=4096, llen=8192>
+  extent4 <loffs = 4096, boffs = 100, len  = 10> -> blob3<raw, len_on_disk=4096, llen=4096>
```

Comment:
extent lengths doesn't matter much in this example - both 10 and 100 result the same behavior.